contrib/buildrefactor/src/python/pants/contrib/buildrefactor/peek.py

```diff
@@ -0,0 +1,44 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+#./pants peek --path=src/scala/org/pantsbuild/zinc/analysis --target-name=zinc-analysis
+#./pants peek --path=src/scala/org/pantsbuild/zinc/extractor --target-name=zinc-extractor --line_number=True
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.task.task import Task
+from pants.contrib.buildrefactor.buildozer import Buildozer
+
+
+class Peek(Task):
+    """Peek a build file for a specified target
+
+    line_number: optional flag to print the starting line of the target if found
+    """
+
+    @classmethod
+    def register_options(cls, register):
+        super(Peek, cls).register_options(register)
+
+        register('--line_number', help='Prints the starting line number of the named target.', type=bool, default=False)
+
+    def __init__(self, *args, **kwargs):
+        super(Peek, self).__init__(*args, **kwargs)
+
+        self.options = self.get_options()
+
+    def execute(self):
+        self.print_name()
+
+    def print_name(self):
+        for root in self.context.target_roots:
+            address_spec = root.address.spec
+            print('\n')
+            Buildozer.execute_binary('print name', address_spec, suppress_warnings=True)
+            print('Named target found in BUILD file.')
+
+            if self.options.line_number:
+                print('\n\'{}\' starts on line number:'.format(root.name))
+                Buildozer.execute_binary('print startline', address_spec, suppress_warnings=True)
```

Comment:
If the line number were moved above the `print name` call, would that cause problems? I think that moving it would improve the output.

```
'foo:target' starts on line number:
  foo/BUILD:12
my_target(name='target',
  kwarg=1,
  kwarg2=1
)
```
seems better to me than
```
my_target(name='target',
  kwarg=1,
  kwarg2=1
)
'foo:target' starts on line number:
  foo/BUILD:12
```

Reply:
Updated to incorporate the basics of this proposal - although just the target's name and associated line number is still currently being printed, not the entire rule definition. Depending on the level of complexity we're looking for with the peek functionality, printing the entire rule definition may be possible using the Buildozer command `'print rule'` described here: https://github.com/bazelbuild/buildtools/tree/master/buildozer

Pretty printing is made difficult by the Buildozer executable which operates as a black hole, with command `'print name'` always printing `'/ntarget_name/n'` and commands `'print startline'` and `'print endline'` always printing `'\ninteger_value\n'`.